### PR TITLE
Beta docs restructure migrating databases

### PR DESCRIPTION
**Pull request by klgill**
I restructured the "Migrating databases to the control plane" chapter, which includes the following modules:

- Deploying backend services
- Configuring a Ceph backend
- Creating a Ceph NFS cluster
- Stopping OpenStack services
- Migrating databases to MariaDB instances
- Migrating OVN data

Peer reviewers: Please use the checklist: https://docs.google.com/document/d/1hxKJdhGNqSOTg6WrbYjJaQIWDu7x5cl2QRYJ7G0HFt8/edit#heading=h.o8ezt27iw7kl

### CODE REVIEW

**Review by fao89 — APPROVED** [review]

**Review by jistr — APPROVED** [review]

**Comment by jistr** [comment]
Interesting, i tried rebasing normally and git didn't ask me to resolve any conflicts, despite Github here claiming there is one... @klgill should i push the rebased code so that we can merge?

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 1]
```diff
@@ -61,7 +57,7 @@ SOURCE_GALERA_MEMBERS=(
 SOURCE_DB_ROOT_PASSWORD=$(cat ~/tripleo-standalone-passwords.yaml | grep ' MysqlRootPassword:' | awk -F ': ' '{ print $2; }')
 ----
 
-* Prepare MariaDB copy directory and the adoption helper pod
+* Prepare MariaDB copy directory and the adoption helper pod 
 
 * Create a temporary folder to store adoption helper pod (pick storage requests to fit MySQL database size):
```
Comment (JoanneOFlynn2018) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:62]: In the preview, the yaml file content with the apiVersion stuff needs to be indented. Not part of this PR.

File: docs_user/modules/proc_migrating-ovn-data.adoc [hunk 2]
```diff
@@ -195,15 +182,15 @@ spec:
...
+. Also enable `ovn-controller`:
++
 [source,yaml]
 ----
 oc patch openstackcontrolplane openstack --type=json -p="[{'op': 'remove', 'path': '/spec/ovn/template/ovnController/nodeSelector'}]"
 ----
 
-* Delete the ovn-data pod and persistent volume claim with OVN databases backup (consider making a snapshot of it, before deleting)
-
+. Delete the `ovn-data`pod and persistent volume claim with OVN databases backup (consider making a snapshot of it, before deleting):
```
Comment (JoanneOFlynn2018) [inline docs_user/modules/proc_migrating-ovn-data.adoc:192]: Need a space because it doesn't render out the backticks in the preview

```suggestion
. Delete the `ovn-data` pod and persistent volume claim with OVN databases backup (consider making a snapshot of it, before deleting):
```

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 3]
```diff
@@ -1,34 +1,30 @@
...
+.Prerequisites
 
 * Make sure the previous Adoption steps have been performed successfully.
- ** The OpenStackControlPlane resource must be already created at this point.
- ** Podified MariaDB and RabbitMQ are running. No other podified
-control plane services are running.
- ** Required services specific topology. For more information, see xref:pulling-the-openstack-configuration_{context}[Pulling the OpenStack configuration].
+ ** The `OpenStackControlPlane` resource must be already created.
+ ** The control plane MariaDB and RabbitMQ are running. No other control plane services are running.
+ ** Required services specific topology. For more information, see xref:pulling-the-openstack-configuration_{context}[Pulling the {osp_prev_long} configuration].
```
Comment (JoanneOFlynn2018) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:22]: This link doesn't work in the preview

File: docs_user/modules/proc_stopping-openstack-services.adoc [hunk 4]
```diff
@@ -62,25 +41,39 @@ openstack image list -c ID -c Status |grep -E '\| .+ing \|'
 Also collect the services topology specific configuration before stopping services required to gather it live. You will need it to compare it with the post-adoption values later on. For more information, see xref:pulling-the-openstack-configuration_{context}[Pulling the OpenStack configuration].
```
Comment (JoanneOFlynn2018) [inline docs_user/modules/proc_stopping-openstack-services.adoc:41]: This link doesn't work in the preview

File: docs_user/adoption-attributes.adoc [hunk 5]
```diff
@@ -8,7 +8,10 @@ ifeval::["{build}" == "upstream"]
 :OpenStackShort: OSP
 :OpenShiftShort: OCP
 :osp_curr_ver: Antelope
+:osp_prev_ver: Wallaby
 :OpenStackInstaller: TripleO
+:CephService_first_ref: Manila
+:CephService: Manila
```
Comment (fmount) [inline docs_user/adoption-attributes.adoc:14]: `Manila` is not technically a `CephService`, it's an `OSP` component that has a `Ceph` backend in this case (NFS)

File: docs_user/modules/proc_configuring-a-ceph-backend.adoc [hunk 6]
```diff
@@ -0,0 +1,129 @@
+[id="configuring-a-ceph-backend_{context}"]
+
+= Configuring a Ceph backend
+
+If the original deployment uses a Ceph storage backend for any service
+(e.g. Glance, Cinder, Nova, Manila), the same backend must be used in the
+adopted deployment and custom resources (CRs) must be configured accordingly.
+
+If you use {CephService_first_ref}, on TripleO environments, the CephFS driver in {CephService} is configured to use
```
Comment (fmount) [inline docs_user/modules/proc_configuring-a-ceph-backend.adoc:9]: Manila != CephService, it's an OSP service using Ceph as one of the backends.

File: docs_user/modules/proc_configuring-a-ceph-backend.adoc [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+its own keypair. For convenience, modify the `openstack` user so that you
+can use it across all {osp_prev_long} services.
+
+Using the same user across the services serves two purposes:
+
+* The capabilities of the user required to interact with {CephService}
```
Comment (fmount) [inline docs_user/modules/proc_configuring-a-ceph-backend.adoc:15]: same here

File: docs_user/modules/proc_configuring-a-ceph-backend.adoc [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+became far simpler and hence, more became more secure with RHOSP 18.
```
Comment (fmount) [inline docs_user/modules/proc_configuring-a-ceph-backend.adoc:16]: do we need a variable for `RHOSP 18` ?

File: docs_user/modules/proc_configuring-a-ceph-backend.adoc [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+              - secret:
+                  name: ceph-conf-files
+          mounts:
+          - name: ceph
+            mountPath: "/etc/ceph"
+            readOnly: true
+'
+----
+
+. Configuring some {OpenStackShort} services to use Ceph backend may require
```
Comment (fmount) [inline docs_user/modules/proc_configuring-a-ceph-backend.adoc:123]: This is not true anymore, the only service was Cinder and I think it does not require the FSID anymore in its config

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc [hunk 10]
```diff
@@ -0,0 +1,121 @@
+[id="creating-a-ceph-nfs-cluster_{context}"]
+
+= Creating a Ceph NFS cluster
+
+If you use the Ceph via NFS backend with {CephService_first_ref}, prior to adoption, you must create a new clustered NFS service on the Ceph cluster. This service will replace the standalone, pacemaker-controlled `ceph-nfs` service that was used on {osp_prev_long} {osp_prev_ver}.
```
Comment (fmount) [inline docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:5]: CephService_first_ref <- it's Manila, an OSP service, while `NFS` is the Ceph Service we are using in this guide.

File: docs_user/modules/proc_deploying-backend-services.adoc [hunk 11]
```diff
@@ -1,31 +1,24 @@
...
-
-//kgilliga: The backend services deployment intro will be converted to an assembly. Most of the rest of the module will be converted to a procedure.
-//Check xref context.
-
 = Deploying backend services
 
-The following instructions create OpenStackControlPlane CR with basic
-backend services deployed, and all the OpenStack services disabled.
-This will be the foundation of the podified control plane.
+Create the `OpenStackControlPlane` custom resource (CR) with basic backend services deployed, and all the {osp_prev_long} ({OpenStackShort}) services disabled. This will be the foundation of the control plane.
```
Comment (fmount) [inline docs_user/modules/proc_deploying-backend-services.adoc:5]: same comment here about OSP svcs / Ceph svcs

File: docs_user/modules/proc_deploying-backend-services.adoc [hunk 12]
```diff
@@ -1,31 +1,24 @@
...
 In subsequent steps, you import the original databases and then add
-podified OpenStack control plane services.
+{OpenStackShort} control plane services.
 
-== Prerequisites
+.Prerequisites
 
 * The cloud that you want to adopt is up and running, and it is on the
-OpenStack Wallaby release.
+{OpenStackShort} {osp_prev_ver} release.
```
Comment (fmount) [inline docs_user/modules/proc_deploying-backend-services.adoc:13]: same here

**Review by fmount — CHANGES_REQUESTED** [review]

**Comment by klgill** [comment]
> Interesting, i tried rebasing normally and git didn't ask me to resolve any conflicts, despite Github here claiming there is one... @klgill should i push the rebased code so that we can merge?

@jistr I need to incorporate changes from Joanne and Francesco. I'm not sure if it's best to merge first and then incorporate their changes in a separate PR.

File: docs_user/adoption-attributes.adoc (hunk 5 above)
Comment (klgill) [reply docs_user/adoption-attributes.adoc:14]: Can we change the variable to "CephComponent"?

File: docs_user/modules/proc_configuring-a-ceph-backend.adoc (hunk 8 above)
Comment (klgill) [reply docs_user/modules/proc_configuring-a-ceph-backend.adoc:16]: This should probably be Red Hat OpenStack Services on OpenShift (RHOSO) 18.0. We no longer use RHOSP in 18.0.

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 3 above)
Comment (klgill) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:22]: I will update in a separate PR when I eventually rename the "Pulling configuration" module.

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 13]
```diff
@@ -234,13 +227,13 @@ oc rsh mariadb-copy-data << EOF
 EOF
 ----
 
-== Post-checks
+.Verification
 
 Compare the following outputs with the topology specific configuration.
-For more information, see xref:pulling-the-openstack-configuration_{context}[Pulling the OpenStack configuration].
+For more information, see xref:pulling-the-openstack-configuration_{context}[Pulling the {osp_prev_long} configuration].
```
[the remark this answers is not in the document]
Comment (klgill) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:233]: I will update in a separate PR when I eventually rename the "Pulling configuration" module.

File: docs_user/modules/proc_stopping-openstack-services.adoc (hunk 4 above)
Comment (klgill) [reply docs_user/modules/proc_stopping-openstack-services.adoc:41]: I will update in a separate PR when I eventually rename the "Pulling configuration" module.

File: docs_user/modules/proc_configuring-a-ceph-backend.adoc (hunk 9 above)
Comment (klgill) [reply docs_user/modules/proc_configuring-a-ceph-backend.adoc:123]: Do you want me to delete this entire step?

File: docs_user/modules/proc_deploying-backend-services.adoc (hunk 12 above)
Comment (klgill) [reply docs_user/modules/proc_deploying-backend-services.adoc:13]: I think in this case we want the sentence to say "RHOSP 17.1", right? The customer's environment needs to be running RHOSP 17.1 before they begin deploying backend services.

File: docs_user/modules/proc_configuring-a-ceph-backend.adoc (hunk 9 above)
Comment (fmount) [reply docs_user/modules/proc_configuring-a-ceph-backend.adoc:123]: yeah, make sense to not have this statement anymore (/cc @jistr ) (thank you)

File: docs_user/modules/proc_configuring-a-ceph-backend.adoc (hunk 8 above)
Comment (klgill) [reply docs_user/modules/proc_configuring-a-ceph-backend.adoc:16]: Fixed.

File: docs_user/modules/proc_configuring-a-ceph-backend.adoc (hunk 9 above)
Comment (klgill) [reply docs_user/modules/proc_configuring-a-ceph-backend.adoc:123]: Removed.

File: docs_user/adoption-attributes.adoc (hunk 5 above)
Comment (fmount) [reply docs_user/adoption-attributes.adoc:15]: ack, makes sense to me to distinguish between the two names using a variable. Can we use something like: `rhos_component_storage_file`?
- `:rhos_comp_storage_file: Manila` if upstream
- `:rhos_comp_storage_file: Shared File System service` if downstream

File: docs_user/adoption-attributes.adoc [hunk 14]
```diff
@@ -5,10 +5,14 @@
 ifeval::["{build}" == "upstream"]
 :OpenShift: OpenShift
 :osp_prev_long: OpenStack
+:osp_acro: OSP
 :OpenStackShort: OSP
 :OpenShiftShort: OCP
 :osp_curr_ver: Antelope
+:osp_prev_ver: Wallaby
 :OpenStackInstaller: TripleO
+:CephService_first_ref: Manila
```
Comment (fmount) [inline docs_user/adoption-attributes.adoc:14]: for the upstream version, using two variables w/ the same name can be optimize w/ a single variable: basically `rhos_component_storage_file` and `rhos_component_storage_file_first_ref` can be converged to `rhos_component_storage_file`.
@klgill not sure if I'm messing up w/ the pattern you're trying to establish but it was just out of curiosity, I'm ok to keep them separated if it makes the flow cleaner.

File: docs_user/adoption-attributes.adoc [hunk 15]
```diff
@@ -5,10 +5,14 @@
 ifeval::["{build}" == "upstream"]
 :OpenShift: OpenShift
 :osp_prev_long: OpenStack
+:osp_acro: OSP
 :OpenStackShort: OSP
 :OpenShiftShort: OCP
 :osp_curr_ver: Antelope
+:osp_prev_ver: Wallaby
 :OpenStackInstaller: TripleO
```
Comment (fmount) [inline docs_user/adoption-attributes.adoc:13]: this should be `OpenStackPreviousInstaller: TripleO`

Comment (klgill) [reply docs_user/adoption-attributes.adoc:13]: Done.

File: docs_user/adoption-attributes.adoc (hunk 14 above)
Comment (klgill) [reply docs_user/adoption-attributes.adoc:14]: I would love to consolidate these, but the issue is with the first reference of the service in the downstream docs. For example, I could set the value for the "rhos_component_storage_file" to "Manila" for upstream and "Shared File System service (manila)" for downstream, but we only need "(manila)" on the first reference in the downstream documentation. 

On the other hand, if I use the variable and manually type "(manila)" after it, the docs will render as "Manila (manila)" upstream.

I'll ask my team for advice on this.

**Comment by klgill** [comment]
Merging today. Anything that wasn't quite resolved in this PR (or any other PR) is being recorded here: https://issues.redhat.com/browse/RHOSPDOC-1821
